File: 0045-jump-game-ii/0045-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int res = 0, left = 0, right = 0, max_jump;
        while(right < nums.length - 1) {
            max_jump = 0;
            for(int i = left; i <= right; ++i)
                max_jump = Math.max(max_jump, i + nums[i]);
            left = right + 1;
            right = max_jump;
            ++res;
        }
        return res;
    }
}

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        dp = [0 for n in nums]
        for i in range(len(nums) - 2, -1, -1):
            reachable = dp[i+1:i+1+nums[i]]
            if reachable:
                dp[i] = min(reachable) + 1
            else:
                dp[i] = math.inf
        return dp[0]